index.js

```diff
@@ -0,0 +1,134 @@
+        }
+}
+
+//Validando los links y obteniendo estadísticos
+function validateAndStats(links){ 
+    let promises = links.map(link => fetch(link)
+        .then (res => {
+        let resp = {
+            url: res.url,
+            status: res.status,
+            text: res.statusText
+        }
+        if(resp.status === 200){
+            console.log(colors.magenta(res.status), colors.cyan(res.statusText), ('url:', link))
+        }else{
+            console.log(colors.yellow(res.status), colors.green(res.statusText),('url:', link))
+        }
+        return resp
+    })
+    .catch (error => {
+        let err = {
+            url: link,
+            status: 'error',
+            text: error.errno
+        }
+        if(err.status === 'error'){
+            console.log(colors.yellow(err.status), colors.green(err.text), ('url:', link))
+        }
+        return err
+    })
+    )
+    return Promise.all(promises)
+    .then(res => statsLinksValidate(res))
+}
+
+function statsLinksValidate (res){
+    console.log(colors.rainbow('Total links: '), (res.length))
+    console.log(colors.magenta.bold('Broken links: ', res.reduce((accountant, element) => {
+        if (element.status !== 200){
+            return accountant += 1
+          }
+          return accountant
+        },0)))
+    console.log(colors.green.bold('Functional links: ', res.reduce((accountant, elemento) => {
+        if (elemento.status === 200){
+            return accountant += 1
+          }
+          return accountant
+        },0)))
+
+    return res
+}
+
+//Obteniendo solamente los links
+function onlyValidate(links){ 
+    let promises = links.map(link => fetch(link)
+        .then (res => {
+        let resp = {
+            url: res.url,
+            status: res.status,
+            text: res.statusText
+        }
+            if(resp.status === 200){
+                console.log(colors.magenta(res.status), colors.cyan(res.statusText), ('url:', link))
+            }else{
+                console.log(colors.yellow(res.status), colors.green(res.statusText),('url:', link))
+            }
+            return resp
+    })  
+    .catch (error => {
+        let err = {
+            url: link,
+            status: 'error',
+            text: error.errno
+        }
+            if(err.status === 'error'){
+                console.log(colors.yellow(err.status), colors.green(err.text), ('url:', link))
+            }
+            return err
+    })
+    )
+    return Promise.all(promises)
+}
+
+//Obteniendo sólo los estadísticos
+function onlyStats(links){ 
+    let promises = links.map(link => fetch(link)
+   .then(res => ({url: link, status: res.status, text:res.statusText}))
+   .catch(err=>({url: link, status: 'error', error:err.message}))
+   )
+    return Promise.all(promises)
+    //Todos los resultados, para sacar stats, hay que contarlos
+    .then(res => {statsLinksValidate(res)})
+}
+
+
+
+module.exports = {
+    readFile,
+    validateFile
```

Comment:
si no se utiliza no la exportes y la función readFile debe tener una mejor definición(nombre)
```suggestion
    integrationFlags
```